### PR TITLE
profile menu started

### PR DESCRIPTION
**Pull request by belljohan3**

### CODE REVIEW

File: components/AppToolbar.vue [hunk 1]
```diff
@@ -1,9 +1,14 @@
+<!-- eslint-disable @intlify/vue-i18n/no-raw-text -->
 <template>
   <header
-    class="py-4 px-6 border-b border-slate-100 bg-white/80 backdrop-blur-md sticky top-0"
+    class="flex justify-between py-4 px-6 border-b border-slate-100 bg-white/80 backdrop-blur-md sticky top-0"
   >
     <strong v-if="toolbarTitleStore().toolbarTitle">
       {{ toolbarTitleStore().toolbarTitle }}
     </strong>
+    <light class="flex">
```
Comment (sifferhans) [inline components/AppToolbar.vue:9]: I dont think there is a HTML element called `light` :) Did you want the text to be a light font? If so, use a span instead with the class `font-light`

File: components/DropdownMenu.vue [hunk 2]
```diff
@@ -56,14 +56,14 @@ function menuItemClick(event: Event, item: DropdownMenuItem) {
         </p>
         <p
           v-else
-          @click="(event: MouseEvent) => event.stopPropagation()"
           class="flex justify-start items-center gap-1 py-2 px-3"
+          @click="(event: MouseEvent) => event.stopPropagation()"
```
Comment (sifferhans) [inline components/DropdownMenu.vue:60]: What is the `stopPropagation` for?

File: components/DropdownMenu.vue [hunk 3]
```diff
@@ -56,14 +56,14 @@ function menuItemClick(event: Event, item: DropdownMenuItem) {
...
         >
           <IconComponent v-if="item.icon" :name="item.icon" />
           <span>{{ item.text }}</span>
         </p>
       </li>
     </ul>
-    <div @click="close" class="fixed inset-0 z-10"></div>
+    <div class="fixed inset-0 z-10" @click="close"></div>
```
Comment (sifferhans) [inline components/DropdownMenu.vue:67]: Can we use a button here instead of a div? Since we have a click handler here it makes sense to use the semantic HTML element

File: layouts/default.vue [hunk 4]
```diff
@@ -1,9 +1,7 @@
 <script lang="ts" setup>
 import { MediaPlaylistInjectionKey } from "~/plugins/3.mediaPlayer";
 
-const onError = (error: any) => {
-  console.error(error);
-};
+const onError = (error: any) => error;
```
Comment (sifferhans) [inline layouts/default.vue:4]: What is the reason for this change? 
With this change the `onError` function only returns the error we get from the `NuxtErrorBoundary`, but we dont do anything with it

**Review by sifferhans — CHANGES_REQUESTED** [review]

File: components/DropdownMenu.vue (hunk 2 above)
Comment (belljohan3) [reply components/DropdownMenu.vue:60]: I don't remember that i made changes in this file.

File: layouts/default.vue (hunk 4 above)
Comment (belljohan3) [reply layouts/default.vue:4]: I made these changes to avoid getting the workflow errors! when I do the `git push`

File: components/DropdownMenu.vue (hunk 2 above)
Comment (belljohan3) [reply components/DropdownMenu.vue:60]: I think it is @bjornarhagen who worked on that !!

**Comment by SimonSimCity** [comment]
@sifferhans IMO some of your comments should have gone to a different PR. Most of the changes here were sorely done to reduce the amount of warnings eslint spits out, mostly https://eslint.vuejs.org/rules/attributes-order.html.

Because you asked: The event propagation is needed for the `click` event not to bubble up to the `track` item, which the `dropdownMenu` component is a part of, which in turn has a different `click` event. See https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation for further information. IMO, the function `stopPropagation()` shouldn't be called inside a function, because its then hidden from the HTML-structure because of which it is called. IMO, we should take a turn on all events and call `event.stopPropagation(); myFunction(event)` if possible, to know where the propagation is done and where its not done. Sometimes its also unwanted to stop the propagation - and in those cases its better to have the code where its important to know about the side-effects. But this is more of me going into the nature of functional programming. Comments welcome 🤗

File: components/AppToolbar.vue [hunk 5]
```diff
@@ -1,9 +1,14 @@
+<!-- eslint-disable @intlify/vue-i18n/no-raw-text -->
```
Comment (SimonSimCity) [inline components/AppToolbar.vue:1]: I think you added this because eslint started to complain about something in this component. Please avoid using disabling of rules, if its not strictly wanted to leave them out. This rule should just create a warning, no error, so you should be able to deploy just fine.

To us, having a warning here, shows that there's still some work to do - in your case, there's text which should be translated. By having this eslint rule disabled, we have to technical mechanism of that there are things still not final.

File: layouts/default.vue (hunk 4 above)
Comment (SimonSimCity) [reply layouts/default.vue:4]: Please revert these changes. They are reported as warning by eslint because we use `console`, but its not an error and this shouldn't stop you from contributing.

@sifferhans Related to https://github.com/bcc-code/bmm-web/issues/33 - we should define a strategy of how to deal with errors of various kinds.

**Review by SimonSimCity — CHANGES_REQUESTED** [review]
Just these three changes and I'm fine:
- eslint rule
- `light` tag
- error-handling of `NuxtErrorBoundary`

File: layouts/default.vue (hunk 4 above)
Comment (sifferhans) [reply layouts/default.vue:4]: Nuxt has a `NuxtErrorBoundary` component that catches errors from child components, which has a `@error` event. We could look into using this for client side errors, but we probably need a separate strategy for API errors

Comment (SimonSimCity) [reply layouts/default.vue:4]: @sifferhans See https://github.com/bcc-code/bmm-web/issues/33#issuecomment-1531143942. We should move this discussion, its off topic here.

File: components/AppToolbar.vue (hunk 5 above)
Comment (belljohan3) [reply components/AppToolbar.vue:1]: Ok. Which text should be translated please ?

File: package.json [hunk 6]
```diff
@@ -19,8 +19,12 @@
   },
   "dependencies": {
     "@auth0/auth0-vue": "^2.2.0",
+<<<<<<< HEAD
```
Comment (SimonSimCity) [inline package.json:22]: Seems like you have not resolved all the merge-conflicts 😅

**Review by SimonSimCity — CHANGES_REQUESTED** [review]
Would be nice to get this PR merged soon. Please take a look at the open points and discuss or change where necessary.

File: pnpm-lock.yaml [hunk 7]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (belljohan3) [reply pnpm-lock.yaml:1]: The conflict has been resolved now.

File: package.json (hunk 6 above)
Comment (belljohan3) [reply package.json:22]: Yes, i have to resolve it.

File: components/ProfileMenu.vue [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        enter-to-class="transform scale-100 opacity-100"
+        leave-active-class="transition duration-75 ease-in"
+        leave-from-class="transform scale-100 opacity-100"
+        leave-to-class="transform scale-95 opacity-0"
+      >
+        <MenuItems
+          class="absolute right-0 mt-2 w-56 origin-top-right divide-y divide-gray-100 rounded-md bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none"
+        >
+          <div class="px-1 py-1">
+            <MenuItem v-slot="{ active }">
```
Comment (SimonSimCity) [inline components/ProfileMenu.vue:29]: All those menu items seem to have the same property. Can't we use a loop here..?

I'd only include options which are done yet or show some kind of message - not something which we are about to implement.

**Review by SimonSimCity — CHANGES_REQUESTED** [review]

File: components/ProfileMenu.vue (hunk 8 above)
Comment (belljohan3) [reply components/ProfileMenu.vue:29]: I just look at the headlessui documentation. I would like also to refact it to have less code. Would you let me know how to do it ? 
-------------------
Also, i've installed nuxt-headlessui module and set it up in nuxt.config.ts file. But i am still getting red line when i am trying to import elements i need for the menu
